### PR TITLE
Added abillity to hide LootTables and PDC

### PR DESCRIPTION
**Pull request by GOjayson**
Added this for #8982

Have closed a PR I had made before because i messed up the Branch in some way.

### CODE REVIEW

File: patches/server/0969-Added-config-option-to-hide-Loottables-and-PDC.patch [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+     public CompoundTag sanitizeSentNbt(CompoundTag tag) {
+-        tag.remove("PublicBukkitValues");
++
++        if(level.paperConfig().anticheat.obfuscation.entitydata.hideLootTables && tag.contains("LootTable")){
++            tag.remove("LootTable");
++            tag.remove("LootTableSeed");
++        };
++        if(level.paperConfig().anticheat.obfuscation.entitydata.hidePDC && tag.contains("PublicBukkitValues") ){
++            tag.remove("PublicBukkitValues");
++        };
```
Comment (Machine-Maker) [inline patches/server/0969-Added-config-option-to-hide-Loottables-and-PDC.patch]: Do you need the extra contains() checks? remove should still succeed if it doesn't have that tag key. It's just one less lookup

**Review by Machine-Maker — APPROVED** [review]
Approved, but the merger has to put stuff in the right patch files and fix the formatting on the conditional blocks. Also verify that `hidePDC` doesn't turn into `hide-p-d-c` in the config file.

File: patches/server/0969-Added-config-option-to-hide-Loottables-and-PDC.patch (hunk 1 above)
Comment (GOjayson) [reply patches/server/0969-Added-config-option-to-hide-Loottables-and-PDC.patch]: I just added this just to be sure. I will test it with the rest.

**Comment by GOjayson** [comment]
Ah yes, it changed to hide-p-d-c

image

as for the contains() they are not needed. Will update the patch now.

**Review by Machine-Maker — APPROVED** [review]
Merger must move the diff for the `WorldConfiguration` class into the `Paper Config` patch and fix the formatting on the conditionals in the other hunk.

**Comment by Owen1212055** [comment]
Had to change to pdc, setting to PDC results in 
image

**Review by Owen1212055 — APPROVED** [review]

**Review by Owen1212055 — CHANGES_REQUESTED** [review]
Looking again, it doesn't seem that there are any cases where block entities will actually send their loot table information.

Certain block entities have to run their serialization logic to pass extra data to the client, and this is what causes PDC to be sent with it. So, I don't actually see why we should make this configurable, as this is unreliable behavior that should most likely be fixed by moving the core serialization behavior.

It would be better to move the PDC serialization logic into one of the different block entity methods instead.

But this is out of the scope of this PR, it's more that the issue this is based off is misleading, and if it's actually worth exposing this as a config option (since this is actually a small side effect of an implementation detail)

**Comment by GOjayson** [comment]
> Looking again, it doesn't seem that there are any cases where block entities will actually send their loot table information.
> 
> Certain block entities have to run their serialization logic to pass extra data to the client, and this is what causes PDC to be sent with it. So, I don't actually see why we should make this configurable, as this is unreliable behavior that should most likely be fixed by moving the core serialization behavior.
> 
> It would be better to move the PDC serialization logic into one of the different block entity methods instead.
> 
> But this is out of the scope of this PR, it's more that the issue this is based off is misleading, and if it's actually worth exposing this as a config option (since this is actually a small side effect of an implementation detail)

So I should undo the PDC part? leave it as it was before?

**Comment by Owen1212055** [comment]
I am not sure, I feel like it may actually be best to leave this not configurable. Because to me, this is a bug.

**Comment by GOjayson** [comment]
Closing PR since issue is closed for being wrong.

**Pull request closed without merging**